huaweicloud/services/cae/data_source_huaweicloud_cae_environments.go

```diff
@@ -0,0 +1,196 @@
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"environment_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the ID of the environment.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the name of the environment.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the status of the environment.",
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the enterprise project ID of the environment.",
+			},
+
+			// attributes
+			"environment_items": {
+				Type:        schema.TypeList,
+				Elem:        environmentItem(),
+				Computed:    true,
+				Description: "The items list of the environment.",
+			},
+		},
+	}
+}
+
+func environmentItem() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the environment item.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the environment item.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the environment item, enum: `creating`、`finish`、`deleting`、`freeze`、`police_freeze`、`delete_failed`.",
+			},
+			"annotations": {
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+				Description: "The additional attributes of the environment item.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The created time of the environment item.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The updated time of the environment item.",
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourceCaeEnvironmentRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// listEnvironment: Query the list of CAE environment
+	var (
+		listCaeEnvironmentsHttpUrl = "v1/{project_id}/cae/environments"
+		listCaeEnvironmentsProduct = "cae"
+	)
+	listCaeEnvironmentsClient, err := cfg.NewServiceClient(listCaeEnvironmentsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CAE client: %s", err)
+	}
+
+	listCaeEnvironmentsPath := listCaeEnvironmentsClient.Endpoint + listCaeEnvironmentsHttpUrl
+	listCaeEnvironmentsPath = strings.ReplaceAll(listCaeEnvironmentsPath, "{project_id}", listCaeEnvironmentsClient.ProjectID)
+
+	listCaeEnvironmentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
```

Comment:
```suggestion
```
using default ok codes to avoid the undefined codes, such as `202`

Reply:
done